src/main/scala/com/thirdship/libunit/CompoundTSUnit.scala

```diff
@@ -165,17 +165,22 @@ class CompoundTSUnit(val numerator: List[TSUnit] = List.empty[TSUnit],
 		}
 	}
 
+	private def keySetEquals(unit: CompoundTSUnit): Boolean = {
+		numeratorDimensions.keySet.equals(unit.numeratorDimensions.keySet) &&
```

Comment:
Good point. Turns out there is a similar method that equals uses: equateUnitLists.
This looks like a more rugged version of keySetEquals.